job/server/src/main/java/alluxio/master/job/plan/PlanCoordinator.java

```diff
@@ -197,17 +200,24 @@ public void setJobAsFailed(String errorMessage) {
    */
   public void failTasksForWorker(long workerId) {
     synchronized (mPlanInfo) {
-      Long taskId = mWorkerIdToTaskId.get(workerId);
-      if (taskId == null) {
+      List<Long> taskIds = mWorkerIdToTaskIds.get(workerId);
+      if (taskIds == null) {
         return;
       }
-      TaskInfo taskInfo = mPlanInfo.getTaskInfo(taskId);
-      if (taskInfo.getStatus().isFinished()) {
-        return;
+
+      boolean statusChanged = false;
+      for (Long taskId : taskIds) {
+        TaskInfo taskInfo = mPlanInfo.getTaskInfo(taskId);
+        if (taskInfo.getStatus().isFinished()) {
```

Comment:
is it always guaranteed that if one taskInfo is finished that we don't want to fail the rest?

Reply:
You are right. This is wrong.

Reply:
I naively left the code but clearly that's not the right logic when there are multiple tasks